src/plasmapy/plasma/grids.py

```diff
@@ -1107,6 +1107,51 @@ def vector_intersects(self, p1, p2):
 
         return Tmin < Tmax
 
+    def soften_edges(self, width: u.Quantity | None = None) -> None:
```

Comment:
This function exists to help get rid of some of those "Quantities at the edges of the grid should go to zero..." warnings the particle tracker always throws. If the grid has a sharp edge, that gradient can mess up the tracking results, so this is a nice pre-filter to run before particle tracing.